### PR TITLE
New random Function

### PR DESCRIPTION
**Pull request by MaxBoehler**
Possibility to choose between absolute and relative Random.

Setting 
`<absRandom> true </absRandom>`
in XML, activates absolute random numbers between (-range, +range).

Setting 
`<absRandom> false </absRandom>`
in XML, activates relative random numbers between  (T *(1-range), T *(1+range)).

It also supports floating range!

### CODE REVIEW

File: src/interfaces/SolverI.cpp [hunk 1]
```diff
@@ -346,20 +346,33 @@ void SolverI::Init() {
             m_string_solver == SolverTypes::NSTempTurbSolver) {
             // Random temperature
             real Ta = params->getReal("initial_conditions/Ta");        // ambient temperature in KELVIN!
-            real A = params->getReal("initial_conditions/A");            // amplitude
-            size_t range = static_cast<size_t>(params->getInt("initial_conditions/range")); // range of random numbers (0-range)
-            Functions::Random(T0, Ta, A, range);
+
+            bool absCheck = params->getBool("initial_conditions/absRandom");
+            if (absCheck == true) {
```
Comment (LinhWuerzburger) [inline src/interfaces/SolverI.cpp:351]: this should be simplified to if (absCheck)

**Comment by LinhWuerzburger** [comment]
there may be a newer version of a random generator in the newer c+ standards (11,17), if it's thread-safe as well, this may be a better/alternative approach

**Comment by MaxBoehler** [comment]
Updated Code:
ToDo: Set interval
ToDo: Set seed

**Comment by LinhWuerzburger** [comment]
random function does not have to be thread safe, because it won't be parallised

**Comment by MaxBoehler** [comment]
Example input:

```
<initial_conditions usr_fct = "Uniform">  
    <val> 200 </val>
    <random> true </random>
    <customSeed> false </customSeed>
    <customSteps> false </customSteps>
    <absolute> true </absolute>
    <range> 10 </range>
</initial_conditions>
```

**Comment by MaxBoehler** [comment]
Alternate Input with stepsize and seed:

```
<initial_conditions usr_fct = "Uniform"> 
    <val> 200 </val>
    <random> true </random>
    <customSeed> true </customSeed>
    <seed> 13 </seed>
    <customSteps> true </customSteps>
    <stepsize> 0.1 </stepsize>
    <absolute> true </absolute>
    <range> 10 </range>
</initial_conditions>
```

**Comment by LinhWuerzburger** [comment]
I'll review it tomorrow :+1:

**Comment by LinhWuerzburger** [comment]
untested yet, will do it tomorrow

**Comment by LinhWuerzburger** [comment]
proposed xml input as implemented:

```
<initial_conditions usr_fct="whatever" random="Yes">
     <random custom_steps="Yes" absolute="Yes" custom_seed="Yes">
         <seed> 13 </seed>                 <!-- mandatory if custom_seed="Yes" -->
         <step_size> 0.1 </step_size> <!-- mandatory if custom_steps="Yes" -->
         <range> 10 </range>
         <value> 200 </value>
     </random>
     <!-- stuff from initial condition -->
</initial_conditions>
```

File: src/utility/Parameters.cpp [hunk 2]
```diff
@@ -6,6 +6,7 @@
 
 #include <iostream>
 #include <vector>
+#include <sstream>
```
Comment (LinhWuerzburger) [inline src/utility/Parameters.cpp:9]: remains of the getBool function?

File: src/Functions.h [hunk 3]
```diff
@@ -19,6 +19,7 @@ struct FunctionNames{
     static const std::string GaussBubble;
     static const std::string Hat;
     static const std::string McDermott;
+    static const std::string RandomC;
     static const std::string RandomT;
```
Comment (LinhWuerzburger) [inline src/Functions.h:23]: RandomT can be removed? at least I think it shouldn't be used anymore

File: src/Functions.cpp [hunk 4]
```diff
@@ -1122,16 +1121,16 @@ namespace Functions {
         return result;
     }
 
-// ============== Temperature initial condition with random distribution =================
+// === Random function with field as ambient value (e.g. for superposition of temperature layers and random values)
```
Comment (LinhWuerzburger) [inline src/Functions.cpp:1124]: the description needs an update

File: src/Functions.cpp [hunk 5]
```diff
@@ -1122,16 +1121,16 @@ namespace Functions {
...
 // ***************************************************************************************
 /// \brief  Random function for any field
-/// \param  out		temperature
-/// \param 	Va 		ambient value
-/// \param  A		amplitude of random numbers
-/// \param	range	range of random numbers
+/// \param  out     field
```
Comment (LinhWuerzburger) [inline src/Functions.cpp:1127]: documentation is not up to date

**Pull request merged**